src/main/java/com/conveyal/datatools/common/utils/NonRuntimeAWSException.java

```diff
@@ -0,0 +1,21 @@
+package com.conveyal.datatools.common.utils;
+
+import com.amazonaws.services.ec2.model.AmazonEC2Exception;
+
+/**
+ * A helper exception class that does not extend the RunTimeException class in order to make the compiler properly
+ * detect possible places where an exception could occur.
+ */
+public class NonRuntimeAWSException extends Exception {
+    public final Exception originalException;
+
+    public NonRuntimeAWSException(String message) {
+        super(message);
+        originalException = null;
+    }
+
+    public NonRuntimeAWSException(AmazonEC2Exception e) {
```

Comment:
Should there just be one constructor that wraps an `AmazonServiceException`? It seems like that might cover all the cases we need it for (but there may be other reasons for doing it the way you have it here).